lib/srv/usermgmt.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package srv
+
+import (
+	"os/user"
+	"runtime"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	"github.com/siddontang/go/log"
+)
+
+func NewUserManagement() (UserManagement, error) {
+	if runtime.GOOS == "windows" {
+		return nil, trace.NotImplemented("Host user creation management is only supported on linux")
+	}
+	return &unixMgmt{}, nil
+}
+
+type UserManagement interface {
+	GetAllUsers() ([]string, error)
+	UserGIDs(*user.User) ([]string, error)
+	Lookup(string) (*user.User, error)
+	LookupGroup(string) (*user.Group, error)
+	groupAdd(string) (int, error)
+	userAdd(string, []string) (int, error)
+	userDel(string) (int, error)
+}
+
+type userCloser struct {
+	mgmt UserManagement
+	user string
+}
+
+func (u *userCloser) Close() error {
+	teleportGroup, err := u.mgmt.LookupGroup(types.TeleportServiceGroup)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(deleteUserInGroup(u.mgmt, u.user, teleportGroup.Gid))
+}
+
+// todo(lxea): add tests now that there is an interface
+
+func createGroupIfNotExist(mgmt UserManagement, group string) error {
```

Comment:
Like mentioned above, I would turn this into an interface, for example:

```go
type HostUsers interface {
  Create(name string, groups []string) error
  Delete(name string)
}
```

and then plug in your OS-specific implementations (what you call `mgmt` - which is only Unix for now) into it.